blocks.js

```diff
@@ -3484,6 +3484,11 @@ BlockMorph.prototype.destroy = function () {
     this.allComments().forEach(function (comment) {
         comment.destroy();
     });
+    
+    if (this.topBlock() === this && this.activeProcess()) {
```

Comment:
Yep, that's right, we can assume that!

Actually, I thought calling ``topBlock()`` would be the right way because delete is a one-time operation, so this extra time wouldn't pile up. But yeah, it may be a problem in huge scripts. Now that I think about it, that could also be a warning that somebody's script is WAY TOO LONG, so they should be breaking it down into manageable pieces, but that's another discussion ;)

Let me update this commit.